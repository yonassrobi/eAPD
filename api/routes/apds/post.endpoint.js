const { APD_TYPE } = require('@cms-eapd/common');
const {
  getDB,
  setupDB,
  teardownDB,
  login,
  unauthenticatedTest,
  unauthorizedTest
} = require('../../endpoint-tests/utils');

describe('APD endpoint | POST /apds', () => {
  const db = getDB();
  beforeAll(async () => {
    await setupDB(db);
  });
  afterAll(async () => {
    await teardownDB(db);
  });

  const url = `/apds`;

  unauthenticatedTest('post', url);
  unauthorizedTest('post', url);

  it('when authenticated as a user with permission', async () => {
    const api = login();
    const response = await api.post(url, {
      apdType: APD_TYPE.MMIS
    });

    expect(response.status).toEqual(200);

    // The created and updated dates are based on when the APD is saved. Rather
    // than figure out something fancy with the snapshots, just pull out the
    // dates and test them with a regex.
    const { created, updated } = response.data;
    delete response.data.created;
    delete response.data.updated;
    delete response.data.id;

    expect(created).toMatch(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/);
    expect(updated).toMatch(/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/);
    expect(response.data).toMatchSnapshot();
  });
});
