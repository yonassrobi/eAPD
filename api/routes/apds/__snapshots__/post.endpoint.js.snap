// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APD endpoint | POST /apds when authenticated as a user with permission 1`] = `
Object {
  "activities": Array [],
  "apdOverview": Object {},
  "apdType": "MMIS",
  "assurancesAndCompliances": Object {
    "independentVV": Array [
      Object {
        "checked": null,
        "explanation": "",
        "title": "45 CFR Part 95.626",
      },
    ],
    "procurement": Array [
      Object {
        "checked": null,
        "explanation": "",
        "title": "SMM, Part 11",
      },
      Object {
        "checked": null,
        "explanation": "",
        "title": "45 CFR Part 95.615",
      },
      Object {
        "checked": null,
        "explanation": "",
        "title": "45 CFR Part 92.36",
      },
    ],
    "recordsAccess": Array [
      Object {
        "checked": null,
        "explanation": "",
        "title": "42 CFR Part 433.112(b)(5)-(9)",
      },
      Object {
        "checked": null,
        "explanation": "",
        "title": "45 CFR Part 95.615",
      },
      Object {
        "checked": null,
        "explanation": "",
        "title": "SMM Section 11267",
      },
    ],
    "softwareRights": Array [
      Object {
        "checked": null,
        "explanation": "",
        "title": "45 CFR Part 95.617",
      },
      Object {
        "checked": null,
        "explanation": "",
        "title": "42 CFR Part 431.300",
      },
      Object {
        "checked": null,
        "explanation": "",
        "title": "45 CFR Part 164",
      },
    ],
  },
  "keyStatePersonnel": Object {
    "keyPersonnel": Array [],
    "medicaidDirector": Object {
      "email": "",
      "name": "",
      "phone": "",
    },
    "medicaidOffice": Object {
      "address1": "",
      "address2": "",
      "city": "",
      "state": "",
      "zip": "",
    },
  },
  "name": "MMIS IAPD",
  "previousActivities": Object {
    "actualExpenditures": Object {
      "2021": Object {
        "mmis": Object {
          "50": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
          "75": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
          "90": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
        },
      },
      "2022": Object {
        "mmis": Object {
          "50": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
          "75": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
          "90": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
        },
      },
      "2023": Object {
        "mmis": Object {
          "50": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
          "75": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
          "90": Object {
            "federalActual": 0,
            "totalApproved": 0,
          },
        },
      },
    },
    "previousActivitySummary": "",
  },
  "securityPlanning": Object {
    "businessContinuityAndDisasterRecovery": "",
    "securityAndInterfacePlan": "",
  },
  "statePrioritiesAndScope": Object {
    "medicaidProgramAndPriorities": "",
    "mesIntroduction": "",
    "scopeOfAPD": "",
  },
  "yearOptions": Array [
    "2023",
    "2024",
    "2025",
  ],
  "years": Array [
    "2023",
    "2024",
  ],
}
`;
